datafusion/src/datasource/object_store/mod.rs

```diff
@@ -40,18 +41,21 @@ use crate::error::{DataFusionError, Result};
 #[async_trait]
 pub trait ObjectReader: Send + Sync {
     /// Get reader for a part [start, start + length] in the file asynchronously
-    async fn chunk_reader(&self, start: u64, length: usize)
-        -> Result<Box<dyn AsyncRead>>;
+    async fn chunk_reader(
+        &self,
+        start: u64,
+        length: usize,
+    ) -> Result<Box<dyn AsyncRead>, Box<dyn Error>>;
```

Comment:
Shouldn't this be `Send + Sync` also?

Reply:
Probably.  

But it seems a majority of the issues currently are coming from converting `DataFusionError`.  I havent had chance to look into this yet though.

Reply:
Not disagreeing with the overall point of this PR, error mapping is a pretty laborious task in Rust, I'm just trying to save a second PR when/if the Send + Sync bites you. :)

Reply:
aha i hear you and agree - this was more painful than i was hoping.  i will def add those traits if needed. fyi context for the request comes from this https://github.com/datafusion-contrib/datafusion-objectstore-s3/pull/12